src/typeahead/typeahead.directive.ts

```diff
@@ -344,7 +344,15 @@ export class TypeaheadDirective implements OnInit, OnDestroy {
     this._subscriptions.push(
       this.keyUpEventEmitter
         .debounceTime(this.typeaheadWaitMs)
-        .mergeMap(() => this.typeahead)
+        .mergeMap((value: string) => {
+          const normalizedQuery = this.normalizeQuery(value);
+
+          return this.typeahead.map((data: any[]) => {
```

Comment:
`any`? - use types or interfaces
remove `return`s

Reply:
`typeahead` has type any in the whole module. Adding type for this instance should be a separate issue.